### PR TITLE
Fix FTP tests + bug in PhpseclibSftp::fetchKeys()

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by Lctrs** [comment]
@nicolasmure @NiR- It's finally ready to be reviewed o/

**Comment by Lctrs** [comment]
cc @fabschurt

**Comment by Lctrs** [comment]
Also the test suite runs much faster now.

**Comment by nicolasmure** [comment]
Thank you for this win @Lctrs :+1:

File: .travis.yml [hunk 1]
```diff
@@ -39,16 +42,12 @@ before_script:
...
-  - sudo useradd -d /home/./gaufrette gaufrette # Removing "./" from this path would break any further Travis builds
-  - echo "gaufrette:gaufrette" | sudo chpasswd
-  - sudo mkdir /home/gaufrette /srv/empty
-  - sudo chown gaufrette:gaufrette /home/gaufrette /srv/ftp
-  - sudo cp .travis/vsftpd.conf /etc/vsftpd.conf
-  - sudo service vsftpd restart
-  - composer self-update && composer install --prefer-source --no-interaction
-  - if [[ ${TRAVIS_PHP_VERSION:0:1} == "5" ]]; then yes | pecl install ssh2 && composer require herzult/php-ssh; fi
+  - docker run -p ${SFTP_PORT}:22 -d atmoz/sftp:alpine ${SFTP_USER}:${SFTP_PASSWORD}:::${SFTP_BASE_DIR}
+  - docker run -d --name ftpd_server -p ${FTP_PORT}:21 -p 30000-30009:30000-30009 -e "PUBLICHOST=${FTP_HOST}" stilliard/pure-ftpd
```
Comment (NiR-) [inline .travis.yml]: Could you update the `docker-compose` and/or Dockerfiles in order to have exactly the same config locally as Travis please?

Comment (Lctrs) [reply .travis.yml]: I'll do it next week.

**Pull request merged**